### PR TITLE
{Release} Upgrade to Azure CLI 2.17.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,130 @@
...
+
+**AKS **
+
+* `az aks get-credentials`: Clarify documentation for get-credentials (#16011)
+
+**App Service**
+
+* Allow customer to create Python 3.9 function app (#16296)
+* Fixes #14583: az webapp up should generate default name if name isn't provided (#16267)
+* Bugfix: Better error handling when trying to create duplicate ASP in diff location (#16143)
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:38]:
```suggestion
*  Fix: Better error handling when trying to create duplicate ASP in diff location (#16143)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,130 @@
...
+
+* Fix "missing secret or certificate in order to authenticate through a service principal" (#16219)
+
+**Resource**
+
+* `az provider register`: Add --accept-terms for registering RPaaS (#16194)
+
+**resource**
+
+* Fix parsing JSON files with multi-line strings (#15502)
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:103]:
```suggestion
**Resource**

* `az provider register`: Add --accept-terms for registering RPaaS (#16194)
* Fix parsing JSON files with multi-line strings (#15502)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,130 @@
...
+* New license type RHEL_ELS_6 (#16012)
+* Adopt track2 SDK, azure-mgmt-compute==18.0.0 (#15750)
+
+**Container**
+
+* Fix misspelling in `az container create` CLI example text. (#16252)
+
+**DataBoxEdge**
+
+* new command module: support for data-box-edge devices and management (#16193)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:71]:
```suggestion
* New command module: support for data-box-edge devices and management (#16193)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,130 @@
...
+
+**IoT**
+
+* Updated device key generation (#16129)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:75]:
```suggestion
* Update device key generation (#16129)
```

**Review by jiasli — APPROVED** [review]
Core, Profile, Role look good to me.

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,130 @@
...
+
+* Fix parsing JSON files with multi-line strings (#15502)
+
+**Role**
+
+* `az ad sp create-for-rbac`: Deprecate creating Contributor role assignment by default (#16081)
+
+**Security**
+
+* - Add secure score commands (#16198)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:111]:
```suggestion
* Add secure score commands (#16198)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,130 @@
...
+
+**Storage**
+
+* Fix issue #15965: Clarify how to remove multiple legal hold tags with `az storage container legal-hold [clear|set]` (#16167)
+* `az storage account encryption-scope`: GA support (#16270)
+* Fix issue #9959: Trying to download a snapshot version of a file share fails with ResourceNotFound (#16275)
+
+**Synapse**
+
+* 1. Add new cmdlets az synapse sql ad-admin show, create, update, delete (#16241)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:127]:
```suggestion
* Add new cmdlets az synapse sql ad-admin show, create, update, delete (#16241)
```

Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:127]:
```suggestion
* Add new cmdlets az synapse sql ad-admin show, create, update, delete (#16241)
```

File: src/azure-cli/HISTORY.rst (hunk 5 above)
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:111]:
```suggestion
* Add secure score commands (#16198)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,130 @@
 Release History
 ===============
 
+2.17.0
+++++++
+
+**ACR**
+
+* Bump api version. support zone redundancy (#15975)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:11]:
```suggestion
* Bump API version. Support zone redundancy (#15975)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,130 @@
...
+* `az acr scope-map update`: --add and --remove are deprecated, they are renamed to --add-repo --remove-repo (#16238)
+* `az acr scope-map create/update`: Add support to handle Gateway actions. (#16238)
+
+**ACR Token**
+
+* `az acr token create`: support added for gateway actions (#16238)
+
+**AKS**
+
+* fix: add arguments removed by a previous PR (#16080)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:28]:
```suggestion
* Fix: add arguments removed by a previous PR (#16080)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,130 @@
...
+**Monitor**
+
+* `az monitor metrics alert create`: improve error message to give more actionable insight (#16255)
+
+**Network**
+
+* `az network private-endpoint create`: add more declaration of '--subnet' and '--private-connection-resource-id' (#16174)
+* Change validator of application-gateway ssl-cert create (#16256)
+* Migrate network to track2 SDK (#16245)
+* Bugfix for "az network traffic-manager profile create" when using "--routing-method MultiValue" (#16300)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:91]:
```suggestion
* Fix bug for "az network traffic-manager profile create" when using "--routing-method MultiValue" (#16300)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,130 @@
...
+
+* `az acr token create`: support added for gateway actions (#16238)
+
+**AKS**
+
+* fix: add arguments removed by a previous PR (#16080)
+
+**AKS **
+
+* `az aks get-credentials`: Clarify documentation for get-credentials (#16011)
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:32]:
```suggestion
**AKS**

* fix: add arguments removed by a previous PR (#16080)
* `az aks get-credentials`: Clarify documentation for get-credentials (#16011)
```

File: src/azure-cli/HISTORY.rst (hunk 4 above)
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:75]:
```suggestion
* Update device key generation (#16129)
```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,130 @@
...
+
+**App Service**
+
+* Allow customer to create Python 3.9 function app (#16296)
+* Fixes #14583: az webapp up should generate default name if name isn't provided (#16267)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:37]:
```suggestion
* Fix #14583: az webapp up should generate default name if name isn't provided (#16267)
```

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,130 @@
...
+* Bugfix: Better error handling when trying to create duplicate ASP in diff location (#16143)
+
+**ARM**
+
+* `az ts`:  Add support for --tags (#16149)
+* `az ts `: supports deleting a single version (#16295)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:43]:
```suggestion
* `az ts `: Support deleting a single version (#16295)
```

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,130 @@
...
+* `az aro update`: Add RBAC validation on cluster update (#16213)
+* Ensure worker_profile is not None before getting the subnets from (#16309)
+
+**Backup**
+
+* `az backup job list`: Solved -o table bug and added backup_management_type as command input (#16304)
+
+**Batch**
+
+* Upgrades data plane to [azure batch 10.0.0](https://pypi.org/project/azure-batch/10.0.0/) (#16156)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:57]:
```suggestion
* Upgrade data plane to [azure batch 10.0.0](https://pypi.org/project/azure-batch/10.0.0/) (#16156)
```

File: src/azure-cli/HISTORY.rst [hunk 14]
```diff
@@ -3,6 +3,130 @@
...
+
+**ARO**
+
+* `az aro delete`: Add RBAC validation on cluster deletion (#16101)
+* `az aro update`: Add RBAC validation on cluster update (#16213)
+* Ensure worker_profile is not None before getting the subnets from (#16309)
+
+**Backup**
+
+* `az backup job list`: Solved -o table bug and added backup_management_type as command input (#16304)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:53]:
```suggestion
* `az backup job list`: Solve -o table bug and added backup_management_type as command input (#16304)
```

File: src/azure-cli/HISTORY.rst (hunk 12 above)
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:43]:
```suggestion
* `az ts`: supports deleting a single version (#16295)
```

File: src/azure-cli/HISTORY.rst (hunk 13 above)
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:57]:
```suggestion
* Upgrade data plane to [azure batch 10.0.0](https://pypi.org/project/azure-batch/10.0.0/) (#16156)
```

File: src/azure-cli/HISTORY.rst [hunk 15]
```diff
@@ -3,6 +3,130 @@
...
+* new command module: support for data-box-edge devices and management (#16193)
+
+**IoT**
+
+* Updated device key generation (#16129)
+* Update identity-enabled hub tests to fix endpoint RBAC issues (#16128)
+
+**Key Vault**
+
+* `az keyvault key import`: support `--kty` for importing BYOK keys (#16223)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:80]:
```suggestion
* `az keyvault key import`: Support `--kty` for importing BYOK keys (#16223)
```

File: src/azure-cli/HISTORY.rst [hunk 16]
```diff
@@ -3,6 +3,130 @@
...
+
+**Monitor**
+
+* `az monitor metrics alert create`: improve error message to give more actionable insight (#16255)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:84]:
```suggestion
* `az monitor metrics alert create`: Improve error message to give more actionable insight (#16255)
```

File: src/azure-cli/HISTORY.rst [hunk 17]
```diff
@@ -3,6 +3,130 @@
...
+
+**Network**
+
+* `az network private-endpoint create`: add more declaration of '--subnet' and '--private-connection-resource-id' (#16174)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:88]:
```suggestion
* `az network private-endpoint create`: Add more declaration of '--subnet' and '--private-connection-resource-id' (#16174)
```

File: src/azure-cli/HISTORY.rst [hunk 18]
```diff
@@ -3,6 +3,130 @@
...
+* `az acr connected-registry`: New feature for on-prem Azure Container Registry (#16238)
+
+**ACR scope-map**
+
+* `az acr scope-map update`: --add and --remove are deprecated, they are renamed to --add-repo --remove-repo (#16238)
+* `az acr scope-map create/update`: Add support to handle Gateway actions. (#16238)
+
+**ACR Token**
+
+* `az acr token create`: support added for gateway actions (#16238)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:24]:
```suggestion
**ACR**

* Support zone redundancy (#15975)
* `az acr connected-registry`: New feature for on-prem Azure Container Registry (#16238)
* `az acr scope-map update`: --add and --remove are deprecated, they are renamed to --add-repo --remove-repo (#16238)
* `az acr scope-map create/update`: Add support to handle Gateway actions. (#16238)
* `az acr token create`: support added for gateway actions (#16238)
```

**Review by Juliehzl — APPROVED** [review]

**Review by qwordy — APPROVED** [review]

**Pull request merged**